runtime/vm/jvminit.c

```diff
@@ -2452,9 +2452,19 @@ IDATA VMInitStages(J9JavaVM *vm, IDATA stage, void* reserved) {
 #endif /* J9VM_INTERP_ATOMIC_FREE_JNI_USES_FLUSH */
 			TRIGGER_J9HOOK_VM_ABOUT_TO_BOOTSTRAP(vm->hookInterface, vm->mainThread);
 			/* At this point, the decision about which interpreter to use has been made */
-			vm->bytecodeLoop = J9_ARE_ANY_BITS_SET(vm->extendedRuntimeFlags, J9_EXTENDED_RUNTIME_DEBUG_MODE)
-					? (void*)debugBytecodeLoop
-					: (void*)bytecodeLoop;
+			if (J9_ARE_ANY_BITS_SET(vm->extendedRuntimeFlags, J9_EXTENDED_RUNTIME_DEBUG_MODE)) {
+				vm->bytecodeLoop = (void*)debugBytecodeLoop;
+			} else {
+				if (J9VMTHREAD_COMPRESS_OBJECT_REFERENCES(vmThread)) {
```

Comment:
This needs to be `J9JAVAVM_COMPRESS_OBJECT_REFERENCES(vm)`

Reply:
fixed